atef/check.py

```diff
@@ -258,6 +259,11 @@ def compare(self, value: Any, identifier: Optional[str] = None) -> Result:
         if self.invert:
             passed = not passed
 
+        # Some comparisons may be done with array values; require that
+        # all match for a success here:
+        if isinstance(passed, Iterable):
```

Comment:
I ran into this failure locally where `passed` was `np.ndarray([True, False, False])` because I had messed around with the caproto simple IOC used in a sample config. If you try `if np.ndarray([...])` you'll find that it raises and helpfully notes "use `all()`!"

Comparisons against numpy arrays needs to make its way into the test suite itself.

Reply:
On the other hand, this could be the wrong approach.

What about if we detect iterable values prior to running the comparison and reduce them via the configured data reduction method? For example, instead of blindly running `compare([1, 2, 3])` we reduce that to `compare((1 + 2 + 3) / 3)`

Reply:
I think an all match is a reasonable stand-in, but I think probably we'd prefer to move to deliberate reductions (where `all` could be a reduction, somehow?). I might be intending to compare against an array but I also might be trying to target a scalar PV and getting the array by mistake.

Reply:
Very good points. I think you're right - `all` should be a reduction option as well. Probably the default one, for that matter.

If you're OK with it, let's leave this minor change in as a temporary fix and revisit in another PR after further discussion.

Reply:
Absolutely agree on leaving as-is for this PR while we don't have any built-in options for handling array data